contributors/devel/sig-architecture/feature-gates.md

```diff
@@ -0,0 +1,64 @@
+# Feature gate progression in Kubernetes
+
+Feature flags are how Kubernetes tracks progression of functionality that is available to a deployer.
+
+## Lifecycle
+
+Feature progress through `Alpha` -> `Beta` -> `GA`. Sometimes we end up deciding that a feature is not going to be supported and we end up marking them as `Deprecated`.
+
+When we add a feature flag, we basically add if/else conditions to ensure that a feature is ONLY activated when either the default is on or if the deployer has switched it on explicitly.
+
+## Fields in the specification are
+	// Default is the default enablement state for the feature. possible values are true/false.
+	Default bool
+	
+    // LockToDefault indicates that the feature is locked to its default and cannot be changed. possible values are true/false.
+	LockToDefault bool
+	
+    // PreRelease indicates the maturity level of the feature. possible values are "featuregate.Alpha", "featuregate.Beta", "featuregate.GA", "featuregate.Deprecated"
+	PreRelease prerelease
+
+
+## Alpha Features
+
+* `PreRelease` is set to `featuregate.Alpha`
+* `Default` is always set to `false`
+* `LockToDefault` is not set. Defaults to `false`
+
+This enables folks to switch on the feature using the command line. By default it is not switched on.
+
+## Beta Features
+
+* `PreRelease` is set to `featuregate.Beta`
+* `Default` is always set to `true`
+* `LockToDefault` is not set. Defaults to `false`
+
+This enables the feature to be on and available in the default installation of kubernetes. 
+
+Sometimes (rarely) the `Default` is set to `false`. For example when this flag needs some explicit action or support outside of Kubernetes. This tells folks that while this feature is Beta, you still need to do some work to switch it on and use it.
```

Comment:
Would be great if we could add an example here. Maybe the CSIMigration APIs?

https://github.com/kubernetes/kubernetes/blob/4cc1127e9251fff364d5c77e2a9a9c3ad42383ab/pkg/features/kube_features.go#L699

Reply:
I suggest adding more comments if it is locked to false. It is really weried for beginners.